pkg/controller/apmserver/deployment_test.go

```diff
@@ -444,6 +444,30 @@ func TestReconcileApmServer_deploymentParams(t *testing.T) {
 				withInitContainer(),
 			wantErr: false,
 		},
+		{
+			name: "secret token influences checksum",
+			args: args{
+				as:            apmFixture,
+				podSpecParams: defaultPodSpecParams,
+				initialObjects: []runtime.Object{
+					&corev1.Secret{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: certSecretName,
+						},
+					},
+					&corev1.Secret{
```

Comment:
This `Secret` is not used. This is because the token Secret is expected to be read before the call to` deploymentParams( as *apmv1.ApmServer, params PodSpecParams)` and provided as part of `params PodSpecParams`:

```golang
type PodSpecParams struct {
	Version         string
	CustomImageName string

	PodTemplate corev1.PodTemplateSpec

	TokenSecret  corev1.Secret // Here
	ConfigSecret corev1.Secret

	keystoreResources *keystore.Resources
}
```